src/xml.rs

```diff
@@ -27,221 +33,429 @@ pub struct Document {
     ///
     /// The latest version is 1,0,0.
     #[serde(rename = "@Version")]
-    version: String,
+    pub version: String,
+    /// Product
     #[serde(rename = "PRODUCT")]
-    product: Product,
+    pub product: Product,
+    /// Collection of all tracks
     #[serde(rename = "COLLECTION")]
-    collection: Collection,
+    pub collection: Collection,
+    /// Playlist Tree
     #[serde(rename = "PLAYLISTS")]
-    playlists: Playlists,
+    pub playlists: Playlists,
 }
 
+/// Product
 #[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
-struct Product {
+pub struct Product {
     /// Name of product
     ///
     /// This name will be displayed in each application software.
     #[serde(rename = "@Name")]
-    name: String,
+    pub name: String,
     /// Version of application
     #[serde(rename = "@Version")]
-    version: String,
+    pub version: String,
     /// Name of company
     #[serde(rename = "@Company")]
-    company: String,
+    pub company: String,
 }
 
 /// The information of the tracks who are not included in any playlist are unnecessary.
-#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
-struct Collection {
-    /// Number of TRACK in COLLECTION
-    #[serde(rename = "@Entries")]
-    entries: i32,
+#[derive(Debug, PartialEq, Clone, Deserialize)]
+pub struct Collection {
+    // The "Entries" attribute that contains the "Number of TRACK in COLLECTION" is omitted here,
+    // because we can just take the number of elements in the `tracks` vector instead.
+    /// Tracks in the collection.
     #[serde(rename = "TRACK")]
-    track: Vec<Track>,
+    pub tracks: Vec<Track>,
+}
+
+impl Serialize for Collection {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        #[derive(Serialize)]
+        struct Value<'a> {
+            /// Number of TRACK in COLLECTION
+            #[serde(rename = "@Count")]
+            entries: usize,
```

Comment:
The commit message doesn't explain why this is now `@Count` when it was previously `@Entries`. Can you elaborate?

Reply:
The format is inconsistent, and it think I confused the item count for the Collection element ("Entries") with the item count for the Node element ("Count"). The original version was correct. I will revert.

<img width="1126" height="252" alt="image" src="https://github.com/user-attachments/assets/0200c8e2-1f7a-4f84-9c39-af49c0b62862" />